### PR TITLE
gui: package signing changes (bug 1810744)

### PR DESCRIPTION
**Pull request by zzzeid**
- add entitlements/requirements files for codesign
- add app bundle to gzipped tarball
- add script to generate manifest
- add script to generate dmg package

### CODE REVIEW

File: bin/adhoc-sign.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    params = {
+        "windows": {},
+        "macOS": {},
+    }
+
+    for signing_format in signing_formats:
+        if signing_format == "windows":
+            continue
```
Comment (zzzeid) [inline bin/adhoc-sign.py:32]:
```suggestion
        if signing_format == "windows":
            # Skip Windows signing manifest as it is not yet supported in adhoc-signing.
            continue
```

File: README.md [hunk 2]
```diff
@@ -121,3 +121,14 @@ Follow the following conventions for pre-releases:
...
+
+#### Signing and notarizing macOS releases
+
+Uploading the signed artifacts is a manual process at this time. To sign and notarize a macOS release, follow these steps:
+
+- Copy the signing manifest output from the build job.
+- Create a pull request to update `signing-manifests/mozregression-macOS.yml` in the [adhoc-signing](https://github.com/mozilla-releng/adhoc-signing) repo with those changes.
+- Wait for pull request to be merged, and the signing task to finish.
+- After the signing task is finished, download `mozregression-gui-app-bundle.tar.gz` and extract it in `gui/dist`.
+- Run bin/dmgbuild.
```
Comment (zzzeid) [inline README.md:133]:
```suggestion
- Run `./bin/dmgbuild`.
```

File: bin/adhoc-sign.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+    # Windows signing is not yet supported by adhoc-signing for mozregression.
+    signing_formats = {
+        # "windows": ["autograph_authenticode"],
+        "macOS": ["macapp"],
+    }
+
+    params = {}
+
+    for signing_format in signing_formats:
```
Comment (cgsheeh) [inline bin/adhoc-sign.py:28]: Use `for signing_format, app in signing_formats.items()` to avoid `signing_formats[signing_format]` below.

**Review by cgsheeh — CHANGES_REQUESTED** [review]

File: bin/adhoc-sign.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        params[signing_format] = {}
```
[the remark this answers is not in the document]
Comment (zzzeid) [reply bin/adhoc-sign.py]: I don't think that would be any simpler, or cleaner than the existing approach, but I did move things around a little to make the flow a little more logical, and set some of the initial values of the dictionary from the get go.

File: bin/adhoc-sign.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    release = args.release[0]
+    filename = "mozregression-gui-app-bundle.tar.gz"
+    urls = {
+        "macOS": "https://github.com/mozilla/mozregression"
+        f"/releases/download/{release}/{filename}",
+        "windows": "https://github.com/mozilla/mozregression"
+        f"/releases/download/{release}/mozregression-gui.exe",
+    }
+
+    # Windows signing is not yet supported by adhoc-signing for mozregression.
```
[the remark this answers is not in the document]
Comment (zzzeid) [reply bin/adhoc-sign.py]: Removed these lines, but filed [bug 1839227](https://bugzilla.mozilla.org/show_bug.cgi?id=1839227).

File: bin/adhoc-sign.py (hunk 3 above)
Comment (zzzeid) [reply bin/adhoc-sign.py]: Renamed the variables to make more sense.

File: bin/adhoc-sign.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (cgsheeh) [reply bin/adhoc-sign.py]: It looks infinitely better now.

**Review by cgsheeh — APPROVED** [review]

File: bin/adhoc-sign.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (zzzeid) [reply bin/adhoc-sign.py]: Indeed -- likely a remnant from a very early take on this. Good to get rid of the repetition of `params[signing_format]`.

**Pull request merged**